Logging capture causes test errors

When I run a test using the `qtbot` fixture with PySide2, I get the following error

```
============================= test session starts ==============================
platform linux -- Python 3.7.0, pytest-3.7.3, py-1.5.4, pluggy-0.7.1
PySide2 5.11.1 -- Qt runtime 5.11.1 -- Qt compiled 5.11.1
rootdir: /home/user/Documents/Writing/Coding/Python/myproject, inifile:
plugins: qt-3.0.0, pudb-0.6, cov-2.5.1
collected 1 item

test_pytestqt.py E                                                       [100%]

==================================== ERRORS ====================================
_____________________ ERROR at setup of test_preselection ______________________

qapp_args = []

    @pytest.yield_fixture(scope='session')
    def qapp(qapp_args):
        """
        Fixture that instantiates the QApplication instance that will be used by
        the tests.
    
        You can use the ``qapp`` fixture in tests which require a ``QApplication``
        to run, but where you don't need full ``qtbot`` functionality.
        """
        app = qt_api.QApplication.instance()
        if app is None:
            global _qapp_instance
>           _qapp_instance = qt_api.QApplication(qapp_args)

/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/plugin.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py:186: in _handle_with_context
    self._append_new_record(msg_type, message, context=context)
/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py:172: in _append_new_record
    self._records.append(Record(msg_type, message, ignored, context))
/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py:217: in __init__
    self._type_name = self._get_msg_type_name(msg_type)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pytestqt.logging.Record'>
msg_type = PySide2.QtCore.QtMsgType.QtInfoMsg

    @classmethod
    def _get_msg_type_name(cls, msg_type):
        """
            Return a string representation of the given QtMsgType enum
            value.
            """
        if not getattr(cls, '_type_name_map', None):
            cls._type_name_map = {
                qt_api.QtDebugMsg: 'QtDebugMsg',
                qt_api.QtWarningMsg: 'QtWarningMsg',
                qt_api.QtCriticalMsg: 'QtCriticalMsg',
                qt_api.QtFatalMsg: 'QtFatalMsg',
            }
>       return cls._type_name_map[msg_type]
E       KeyError: PySide2.QtCore.QtMsgType.QtInfoMsg

/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py:244: KeyError
=========================== 1 error in 0.17 seconds ============================
```

I get this error even for a test module with these contents:

```
def test_preselection(qtbot):
    pass
```
The output of `pip freeze` is

```
astroid==2.0.4                                                                                                                                                                                                   
atomicwrites==1.2.0                                                                                                                                                                                               
attrs==18.1.0                                                                                                                                                                                        
coverage==4.5.1                                                                                                                                                                                                   
isort==4.3.4                                                                                                                                                                                                      
lazy-object-proxy==1.3.1                                                                                                                                                          
mccabe==0.6.1                                                                                                                                                                                                     
more-itertools==4.3.0                          
mypy==0.620                                                                                                                                                                                                        
pluggy==0.7.1                                                                                                                                                                                                      
pudb==2018.1                                                                                                                                                                                                       
py==1.5.4                                                                                                
pycodestyle==2.4.0                                                                                                                                                                                                
Pygments==2.2.0                                                                                                                                                                                                    
pylint==2.1.1                                                                                                                                                                                                     
PySide2==5.11.1                                                                                                                                                                                                   
pytest==3.7.3                                                                                                                                                                                                     
pytest-cov==2.5.1                                                                                                                                                                                                 
pytest-pudb==0.6                                                                                                                                                                                                   
pytest-qt==3.0.0            
```

Switching from PySide2 to PyQt5, I get a different  error message:

```
===================================================================================================== ERRORS ======================================================================================================
_______________________________________________________________________________________ ERROR at setup of test_preselection _______________________________________________________________________________________
SETUP ERROR: Qt exceptions in virtual methods:
________________________________________________________________________________
Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 186, in _handle_with_context                                                           
    self._append_new_record(msg_type, message, context=context)
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 172, in _append_new_record                                                             
    self._records.append(Record(msg_type, message, ignored, context))
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 217, in __init__                                                                       
    self._type_name = self._get_msg_type_name(msg_type)
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 244, in _get_msg_type_name                                                             
    return cls._type_name_map[msg_type]
KeyError: 4
________________________________________________________________________________

---------------------------------------------------------------------------------------------- Captured stderr setup ----------------------------------------------------------------------------------------------
Qt exceptions in virtual methods:
________________________________________________________________________________
Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 186, in _handle_with_context                                                           
    self._append_new_record(msg_type, message, context=context)
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 172, in _append_new_record                                                             
    self._records.append(Record(msg_type, message, ignored, context))
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 217, in __init__                                                                       
    self._type_name = self._get_msg_type_name(msg_type)
  File "/home/user/.local/share/virtualenvs/myproject-TS3ABiG4/lib/python3.7/site-packages/pytestqt/logging.py", line 244, in _get_msg_type_name                                                             
    return cls._type_name_map[msg_type]
KeyError: 4
```

For both PySide2 and PyQt, the test runs successfully when using `pytest --no-qt-log`.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,3 +1,12 @@
+3.0.1 (unreleased)
+------------------
+
+- Fix handling of ``QtInfoMsg`` objects during logging (`#225`_).
+  Thanks `@willsALMANJ`_ for the report.
+
+.. _#225: https://github.com/pytest-dev/pytest-qt/issues/225
+
+
 3.0.0 (2018-07-12)
 ------------------
 
@@ -496,3 +505,4 @@ First working version.
 .. _@rth: https://github.com/rth
 .. _@Sheeo: https://github.com/Sheeo
 .. _@The-Compiler: https://github.com/The-Compiler
+.. _@willsALMANJ: https://github.com/willsALMANJ

--- a/pytestqt/logging.py
+++ b/pytestqt/logging.py
@@ -241,6 +241,8 @@ class Record(object):
                 qt_api.QtCriticalMsg: 'QtCriticalMsg',
                 qt_api.QtFatalMsg: 'QtFatalMsg',
             }
+            if qt_api.QtInfoMsg is not None:
+                cls._type_name_map[qt_api.QtInfoMsg] = 'QtInfoMsg'
         return cls._type_name_map[msg_type]
 
     @classmethod

--- a/pytestqt/qt_compat.py
+++ b/pytestqt/qt_compat.py
@@ -90,10 +90,13 @@ class _QtApi:
         self.Qt = QtCore.Qt
         self.QEvent = QtCore.QEvent
 
+        # qInfo is not exposed in PyQt5 and PySide2 bindings (#225)
+        self.qInfo = None
         self.qDebug = QtCore.qDebug
         self.qWarning = QtCore.qWarning
         self.qCritical = QtCore.qCritical
         self.qFatal = QtCore.qFatal
+        self.QtInfoMsg = getattr(QtCore, 'QtInfoMsg', None)
         self.QtDebugMsg = QtCore.QtDebugMsg
         self.QtWarningMsg = QtCore.QtWarningMsg
         self.QtCriticalMsg = QtCore.QtCriticalMsg

--- a/tests/test_logging.py
+++ b/tests/test_logging.py
@@ -33,6 +33,8 @@ def test_basic_logging(testdir, test_succeeds, qt_log):
             qt_api.qInstallMsgHandler(print_msg)
 
         def test_types():
+            # qInfo is not exposed by the bindings yet (#225)
+            # qt_api.qInfo('this is an INFO message')
             qt_api.qDebug('this is a DEBUG message')
             qt_api.qWarning('this is a WARNING message')
             qt_api.qCritical('this is a CRITICAL message')
@@ -47,6 +49,8 @@ def test_basic_logging(testdir, test_succeeds, qt_log):
         if qt_log:
             res.stdout.fnmatch_lines([
                 '*-- Captured Qt messages --*',
+                # qInfo is not exposed by the bindings yet (#225)
+                # '*QtInfoMsg: this is an INFO message*',
                 '*QtDebugMsg: this is a DEBUG message*',
                 '*QtWarningMsg: this is a WARNING message*',
                 '*QtCriticalMsg: this is a CRITICAL message*',
@@ -54,6 +58,9 @@ def test_basic_logging(testdir, test_succeeds, qt_log):
         else:
             res.stdout.fnmatch_lines([
                 '*-- Captured stderr call --*',
+                # qInfo is not exposed by the bindings yet (#225)
+                # '*QtInfoMsg: this is an INFO message*',
+                # 'this is an INFO message*',
                 'this is a DEBUG message*',
                 'this is a WARNING message*',
                 'this is a CRITICAL message*',

